src/OrchardCore/OrchardCore.DisplayManagement.Liquid/TagHelpers/LiquidTagHelperActivator.cs

```diff
@@ -52,15 +52,15 @@ public LiquidTagHelperActivator(Type type)
                 }
 
                 var setter = MakeFastPropertySetter(type, property);
+                var viewContextAttribute = property.GetCustomAttribute<ViewContextAttribute>();
+                if (viewContextAttribute != null && property.PropertyType == typeof(ViewContext))
+                {
+                    _viewContextSetter = (helper, context) => setter(helper, context);
+                    continue;
+                }
 
                 foreach (var propertyName in allNames)
                 {
-                    if (propertyName == "ViewContext")
```

Comment:
Code stays here just add condition to compare propertyName ==  viewContentProperty

Reply:
That's why I asked before `allNames` members of  type `HtmlAttributeNameAttribute` I'm not sure if it is by intention or not

Reply:
Ideally property with `[ViewContext]` should also have `[HtmlAttributeNotBound]` so that there is no html binding for such property. So guess your code seems correct.